File: leetcode.com/0728_self_dividing_numbers/solution.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        l = []

        for n in range(left, right + 1):
            for i in map(int, list(str(n))):
                if i == 0:
                    break

                if n % i != 0:
                    break
            else:
                l.append(n)

        return l

File: leetcode.com/0728_self_dividing_numbers/solution.go
func selfDividingNumbers(left int, right int) []int {
    list := make([]int, 0, right - left + 1)

    for n := left ; n <= right ; n++ {

        x := n
        ok := true

        for x > 0 {
            ln := x % 10

            if ln == 0 {
                ok = false
                break
            }

            if n % ln != 0 {
                ok = false
                break
            }

            x /= 10
        }

        if (ok) {
            list = append(list, n)
        }
    }

    return list
}